src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.UnrootedSymbolSet.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal partial class SolutionState
+    {
+        /// <summary>
+        /// In IDE scenarios we have the need to map from an <see cref="ISymbol"/> to the <see cref="Project"/> that
+        /// contained a <see cref="Compilation"/> that could have produced that symbol.  This is especially needed with
+        /// OOP scenarios where we have to communicate to OOP from VS (And vice versa) what symbol we are referring to.
+        /// To do this, we pass along a project where this symbol could be found, and enough information (a <see
+        /// cref="SymbolKey"/>) to resolve that symbol back in that that <see cref="Project"/>.  This is challenging
+        /// however as symbols do not necessarily have back-pointers to <see cref="Compilation"/>s, and as such, we
+        /// can't just see which Project produced the <see cref="Compilation"/> that produced that <see
+        /// cref="ISymbol"/>.  In other words, the <see cref="ISymbol"/> doesn't <c>root</c> the compilation.  Because
```

Comment:
I didn't feel like this sentence added anything.

```suggestion
        /// cref="ISymbol"/>.  Because
```

Reply:
i was trying to give a reason to call it "unrooted symbol set".  i.e. someone might ask "what isn't being rooted".  so i thought i'd be explicit that it was the case that the symbols didn't root the compilation.  Too much?